linux_boot_params/src/lib.rs

```diff
@@ -606,8 +606,8 @@ pub struct EDIDInfo {
     pub dummy: [u8; 128usize],
 }
 
-#[repr(C, packed)]
-#[derive(Copy, Clone)]
+#[repr(C, align(4096))]
```

Comment:
Why did you choose 4096 here?

Reply:
4096 (4 KiB) is the size of a memory page on the x86 architecture, and the expectation is that this data structure is on a separate page of memory. (As a side note, the size of the data structure is also exactly 4 KiB, so it completely fills exactly one page).

In the past, this 4K alignment was enforced as a side effect by the fact that we set up a separate, 4K-aligned section, in the linker script. Now that we no longer do that in the linker script, we need to enforce it here.

We should have had the alignment requirement here from the get-go, the fact that it all worked was just a happy accident.